### PR TITLE
refactor(payment_methods): Remove legacy locker code  as it is not been used

### PR DESCRIPTION
**Pull request by ShankarSinghC**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Remove the legacy locker code from (crates/router/src/core/payment_methods/cards.rs) as we are no more using it. Also there are refactoring of certain functions as the locker_id was only used in case of legacy locker.

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [x] This PR modifies application configuration/environment variables

## How did you test it?

Screenshot 2023-07-13 at 4 15 42 PM

tested on custom pod
Screenshot 2023-07-13 at 4 42 50 PM

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: crates/router/src/configs/defaults.rs [hunk 1]
```diff
@@ -56,7 +56,6 @@ impl Default for super::settings::Locker {
             host: "localhost".into(),
             mock_locker: true,
             basilisk_host: "localhost".into(),
-            locker_setup: super::settings::LockerSetup::LegacyLocker,
```
Comment (SanchithHegde) [inline crates/router/src/configs/defaults.rs:59]: Remove this from the TOML config files as well, if present

**Review by SanchithHegde — COMMENTED** [review]
Other than that, looks good to me!

**Review by SanchithHegde — DISMISSED** [review]

**Comment by Narayanbhat166** [comment]
Why was `locker_id` used before? is it only for Legacy locker or is there some other significance of it? @jarnura

**Comment by ShankarSinghC** [comment]
> Why was `locker_id` used before? is it only for Legacy locker or is there some other significance of it? @jarnura

Currently it's just been used in the case of Legacy locker, in future it might be used for some other purpose also.

**Review by vspecky — DISMISSED** [review]

**Review by inventvenkat — DISMISSED** [review]
Merge it after testing in custom pod in Integ or Sandbox

**Review by Narayanbhat166 — APPROVED** [review]

**Review by SanchithHegde — APPROVED** [review]

**Pull request merged**